packages/redis/test/Adaptor.test.js

```diff
@@ -0,0 +1,334 @@
+import { expect } from 'chai';
+
+import { get, hget, hset, set, scan, setMockClient, hGetAll } from '../src';
+
+describe('get', () => {
+  it('should get the string value of a key', async () => {
+    setMockClient({
+      get: async () => 'john doe',
+    });
+
+    const state = {};
+    const result = await get('patient')(state);
+    expect(result.data).to.eql('john doe');
+  });
+  it('should expand references', async () => {
+    setMockClient({
+      get: async () => 'mtuchi',
+    });
+    const state = { key: 'name' };
+    const result = await get(s => s.key)(state);
+    expect(result.data).to.eql('mtuchi');
+  });
+  it('should return null if key does not exist', async () => {
+    setMockClient({
+      get: async () => null,
+    });
+
+    const state = {};
+    const result = await get('not-found')(state);
+    expect(result.data).to.eql(null);
+  });
+});
+
+describe('hget', () => {
+  it('should return the value of a key and field', async () => {
+    setMockClient({
+      hGet: async () => 'cat',
+    });
+    const state = {};
+    const result = await hget('animals:1', 'species')(state);
+    expect(result.data).to.eql('cat');
+  });
+  it('should return null if field value is not found', async () => {
+    setMockClient({
+      hGet: async () => null,
+    });
+    const state = {};
+    const result = await hget('animals:1', 'cat')(state);
+    expect(result.data).to.eql(null);
+  });
+  it('should throw field value is not specified', async () => {
```

Comment:
I have implemented this for `hget` i would love to implement this in other places, I would wait for your feedback first

Reply:
hmm, well, I still think this is only valuable for hget and maybe hset, because I think the API is a bit weirder. there's more to forget.

We don't usually test arguments in this much detail, and I suppose I can see the value but I'm also not sure that we should be doing it. It feels a bit presumptous and labour intensive.

My instinct is we should only be doing it for very common error cases or very complex arguments.

So I'm going to recommend we just leave it for hget for now.

One thing I would say is: move the logic `asserthgetargs` inside `hget`, but leave `assertArgType` in utils. I definitely don't think we need an assertion for every single operation.